File: highlight/src/main/java/me/rerere/highlight/languages/Languages.kt
package me.rerere.highlight.languages

import me.rerere.highlight.core.Language
import me.rerere.highlight.languages.bash.bash
import me.rerere.highlight.languages.c.c
import me.rerere.highlight.languages.cmake.cmake
import me.rerere.highlight.languages.cpp.cpp
import me.rerere.highlight.languages.css.css
import me.rerere.highlight.languages.dockerfile.dockerfile
import me.rerere.highlight.languages.go.go
import me.rerere.highlight.languages.ini.ini
import me.rerere.highlight.languages.java.java
import me.rerere.highlight.languages.javascript.javascript
import me.rerere.highlight.languages.json.json
import me.rerere.highlight.languages.kotlin.kotlin
import me.rerere.highlight.languages.python.python
import me.rerere.highlight.languages.typescript.typescript
import me.rerere.highlight.languages.xml.xml
import me.rerere.highlight.languages.yaml.yaml

/**
 * Every grammar bundled with the highlighter.
 *
 * Each entry builds a fresh mode tree: compilation mutates modes in place, mirroring `highlight.js`.
 */
internal fun builtinLanguages(): List<Language> = listOf(
    json(),
    ini(),
    cmake(),
    go(),
    yaml(),
    bash(),
    dockerfile(),
    javascript(),
    typescript(),
    xml(),
    css(),
    java(),
    kotlin(),
    python(),
    c(),
    cpp(),
)

File: highlight/tools/languages.mjs
// The grammars bundled by the Kotlin highlighter, with the aliases it answers to.
//
// The fixture generator registers exactly these with highlight.js so that sub-language lookups
// resolve the same way on both sides: a `subLanguage` naming a grammar we do not ship must fall
// back to plain text in the golden fixtures too.

export const LANGUAGES = [
  { name: 'json', module: 'json', aliases: ['json5'] },
  { name: 'ini', module: 'ini', aliases: ['ini'] },
  { name: 'cmake', module: 'cmake', aliases: [] },
  { name: 'go', module: 'go', aliases: [] },
  { name: 'yaml', module: 'yaml', aliases: [] },
  { name: 'bash', module: 'bash', aliases: ['shell'] },
  { name: 'dockerfile', module: 'dockerfile', aliases: [] },
  { name: 'javascript', module: 'javascript', aliases: ['js', 'jsx', 'mjs', 'cjs'] },
  { name: 'typescript', module: 'typescript', aliases: ['ts', 'tsx', 'mts', 'cts'] },
  {
    name: 'xml',
    module: 'xml',
    aliases: ['html', 'xhtml', 'rss', 'atom', 'xjb', 'xsd', 'xsl', 'plist', 'wsf', 'svg'],
  },
  { name: 'css', module: 'css', aliases: [] },
]
